src/Akka.Persistence.Azure/Journal/AzureTableStorageJournal.cs

```diff
@@ -709,39 +544,20 @@ private static TableQuery GenerateAllPersistenceIdsQuery()
 
         private async Task<IEnumerable<string>> GetAllPersistenceIds()
         {
-            var query = GenerateAllPersistenceIdsQuery();
-
-            TableQuerySegment<DynamicTableEntity> result = null;
-
-            var returnValue = ImmutableList<string>.Empty;
-
-            do
-            {
-                result = await Table.ExecuteQuerySegmentedAsync(query, result?.ContinuationToken);
-
-                if (result.Results.Count > 0)
-                {
-                    returnValue = returnValue.AddRange(result.Results.Select(x => x.RowKey));
-                }
-            } while (result.ContinuationToken != null);
-
-            return returnValue;
+            return await GenerateAllPersistenceIdsQuery().Select(item => item.RowKey).ToListAsync();
```

Comment:
With the `IAsyncEnumerator` returned by the new SDK, the old code can be replaced with a single line of code.